src/Service/CheckConsumerPermissionsService.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Enterprise License (PEL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PEL
+ */
+
+namespace Pimcore\Bundle\DataHubBundle\Service;
+
+use Pimcore\Bundle\DataHubBundle\Configuration;
+use Symfony\Component\HttpFoundation\Request;
+
+class CheckConsumerPermissionsService
+{
+    /**
+     * @param Request $request
+     * @param Configuration $configuration
+     *
+     * @return void
+     */
+    public function performSecurityCheck(Request $request, Configuration $configuration): bool
+    {
+        $securityConfig = $configuration->getSecurityConfig();
+        if ($securityConfig['method'] === 'datahub_apikey') {
```

Comment:
I think we should store this constant in a central place. wdyt ?

Reply:
Good Idea. Do you have a class for Constants already? Or should we rather put it as a public const in Configuration.php, that is where it is used anyways, right?